### ISSUE TITLE
JS exception when using mixed case in the domain name of an email address 

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#2891]
STR
1. Sign up as a new user 'examplename@Gmail.com'. Note the upper case 'G' in gmail.
2. Go to an RP, bring up the dialog, and click 'this is not me'.
3. Enter 'examplename@gmail.com' and the password. Note the lower case 'g' in gmail.

Expected: domain names are case-insensitive, so you should be signed in

Actual: javascript exception: 'Error: Error: invalid email' in dialog.js

The back end appears to be fine with either, but haven't thoroughly vetted it.

**Comment by guillaumev** [comment]
Related to this, see https://groups.google.com/forum/?fromgroups=#!topic/mozilla.dev.identity/pa3Sc0kh9Jw and the fact that the website refreshes itself automatically if you are logged in with an "upper case" domain name. jrgm > would you have a test environment for me to see if this happens only on my website (in which case it would be a bug within the Persona drupal module) or if it happens on any persona enabled site ?

Thanks,

**Comment by jrgm** [comment]
We have test RPs at http://123done.org and http://myfavoritebeer.org (the former uses watch/request, while the latter uses get). Or add 'beta.' to point them to use our stage environment as the identity provider login.anosrep.org.

**Comment by jaredhirsch** [comment]
fixed by #3078

**Comment by jrgm** [comment]
Note for testing: this issue was specifically about the domain part of an email address.

Lazy question: @6a68 does #3078 address mixed case on the domain name?

**Comment by jrgm** [comment]
Reopening.

For the original STR, I no longer get  'Error: Error: invalid email' . 

However, using login.dev.anosrep.org + http://dev.123done.org, on every **second** attempt to use the somename@Gmail.com user on dev.123done.org, I get a window.alert 'failed to verify assertion: bad signature in chain'. 

This does not happen in current stage (train-2013.03.01) or on prod (train-2013.02.15). And it appears that the original issue of 'Error: Error: invalid email' was fixed somewhere along the way. 

Anyhow, I'll bump this to 5-star, and we can discuss tomorrow.

**Comment by jaredhirsch** [comment]
jrgm, what browser are you using? I cannot repro this problem on chrome or FF.

**Comment by jrgm** [comment]
I don't recall. Probably FF 19. Let me try again (but I really shouldn't at the moment).

**Comment by jrgm** [comment]
I remember now that the window.alert was from the FF UI. 

But now I cannot reproduce this. Which is really weird since I had gone back and confirmed I could reproduce this last night before making my comment above.

**Comment by jrgm** [comment]
And now I can reproduce this. Or well, the error message in the alert is now for a JSON parse error on the callback for /verify, because the response from 123done.org on the wire is 'An error has occurred: {"code":"ECONNREFUSED","errno":"ECONNREFUSED","syscall":"connect"}'. Is there something wrong with 123done.org? (Which would possibly explain how this issue seems hit and miss). 

![bad-signup-with-Gmail-address](https://f.cloud.github.com/assets/758162/238401/43fa4c64-8836-11e2-9d6a-a04ac15ffd87.png)

**Comment by jaredhirsch** [comment]
umz, can no longer reproduce. twas an artifact of awsbox failytown. quoth the lloydster:

```
lloyd       _6a68: we had two dev instances running on round robin DNS with different keys.
```

**Comment by jrgm** [comment]
Yeah, I can't do this on stage anymore (and dig dev.123done.org is one A record only now).
